specs/hotel_spec.rb

```diff
@@ -0,0 +1,305 @@
+      res_block_setup
+      hotel.reserve_room_in_block("Bob", 3)
+
+      hotel.block_reservations[0].avail_block_rooms.must_be_kind_of Array
+      hotel.block_reservations[0].avail_block_rooms.must_equal [4,5]
+      hotel.block_reservations[0].reserved_rooms.must_be_kind_of Array
+      hotel.block_reservations[0].reserved_rooms.must_equal [1,2,3]
+    end
+
+    it "should update the cost of the reserved rooms from within the block" do
+      res_block_setup
+      hotel.reserve_room_in_block("Bob", 2)
+
+      hotel.block_reservations[0].block_total.must_equal 300
+    end
+
+    it "should raise an ArgumentError if no rooms are available within the block" do
+      res_block_setup
+      hotel.reserve_room_in_block("Bob", 5)
+
+      proc { hotel.reserve_room_in_block("Bob", 5) }.must_raise ArgumentError
+    end
+  end # Describe
+
+  describe "#avail_rooms_in_block" do
+    it "should raise an ArgumentError if the name is not a String" do
+      proc { hotel.avail_rooms_in_block(12345) }.must_raise ArgumentError
+    end
+
+    it "should return an array of available room numbers for a specific block as Integers" do
+      res_block_setup
+      avail_rooms = hotel.avail_rooms_in_block("Bob")
+
+      avail_rooms.must_be_kind_of Array
+      avail_rooms.map { |room_num| room_num.must_be_kind_of Integer }
+    end
+
+    it "should return the first 5 available rooms as apart of the block" do
+      res_block_setup
+
+      hotel.avail_rooms_in_block("Bob").must_be_kind_of Array
+      hotel.avail_rooms_in_block("Bob").length.must_equal 5
+      hotel.avail_rooms_in_block("Bob").must_equal [1,2,3,4,5]
+    end
+
+    it "should return the last 5 available rooms if other blocks are booked for the same days" do
+      hotel.reserve_block("Bob", '2001-02-03', '2001-02-05', 5)
+      hotel.reserve_block("Pete", '2001-02-03', '2001-02-05', 5)
+      hotel.reserve_block("Sally", '2001-02-03', '2001-02-05', 5)
+      hotel.reserve_block("Sue", '2001-02-03', '2001-02-05', 5)
+
+      hotel.avail_rooms_in_block("Sue").must_equal [16,17,18,19,20]
+    end
+
+    it "should raise an ArgumentError if no rooms in the block are available" do
+      res_block_setup
+      hotel.reserve_room_in_block("Bob", 5)
+      proc { hotel.avail_rooms_in_block("Bob") }.must_raise ArgumentError
+    end
+  end
+
+  describe "#all_reservations(date)" do
+    it "should return an array of all Reservations with the requested date" do
+      hotel.create_reservation("Sue", '2001-02-01', '2001-02-10')
+      hotel.create_reservation("Bob", '2001-02-01', '2001-02-03')
+      current_reservations = hotel.all_reservations_on('2001-02-02')
+
+      current_reservations.length.must_equal 2
+      current_reservations.must_be_kind_of Array
+    end
+
+    it "should not return any Reservations if no dates are booked" do
+      hotel.all_reservations_on('2001-02-01').must_be_empty
+    end
```

Comment:
It would also be good to check that reservations for other dates do not affect the result of this method.